src/Storages/WindowView/StorageWindowView.cpp

```diff
@@ -915,26 +915,38 @@ void StorageWindowView::threadFuncCleanup()
 
 void StorageWindowView::threadFuncFireProc()
 {
+    static bool window_kind_larger_than_day = window_kind == IntervalKind::Week || window_kind == IntervalKind::Month
+        || window_kind == IntervalKind::Quarter || window_kind == IntervalKind::Year;
+
     std::unique_lock lock(fire_signal_mutex);
     UInt32 timestamp_now = std::time(nullptr);
 
-    while (next_fire_signal <= timestamp_now)
+    /// When window kind is larger than day, getWindowUpperBound() will get a day num instead of timestamp,
```

Comment:
where do you see it should return daynum?

Reply:
https://github.com/ClickHouse/ClickHouse/blob/11e63f76a105edc21b8f966e13b4c04eeb57ad54/src/Storages/WindowView/StorageWindowView.cpp#L807-L808

`toStartOfTransform<IntervalKind::Week/Month/..>::execute` will get a day num, and `AddTime` also add day num to it.